src/main/java/com/endava/soa_5ed/task2/arrays/CodeReadingTask.java

```diff
@@ -0,0 +1,75 @@
+package com.endava.soa_5ed.task2.arrays;
+
+public class CodeReadingTask {
+    /* Arrays - Exercise 1 */
+    public static int multiplyArrayNumbers(int[] a) {
+        //this method multiplies all the numbers in an array and returns the product
+        int product = 1;
+        int i = 0;
+        while (i < a.length) {
+            product = product * a[i];
+            i++;
+        }
+        return product;
+    }
+
+    public static int checkEquality(int[] a, int numberToCheckAgainst) {
+        //this method checks if the given number is present in the given array and returns it
+        for (int i = 0; i < a.length; i++) {
+            if (a[i] == numberToCheckAgainst) {
+                return a[i];
+            }
+        }
+        return -1;
+    }
+
+    public static int countNumberMatches(int[] a, int numberToCheckAgainst) {
+        //this method counts how many times is the given number present in the given array and returns the count
+        int count = 0;
+        for (int number : a) {
+            if (number == numberToCheckAgainst) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    /*Exercise 2*/
+    public static int[] createArray(int arrayLength) {
+//    this method returns an int array with the given length and the values of all elements equal to 1
+        int[] a = new int[arrayLength];
+        for (int i = 0; i < arrayLength; i++) {
+            a[i] = i + 1;
```

Comment:
Check the task again.

Reply:
I don't understand, could you be more specific, please?